lib/macho/macho_file.rb

```diff
@@ -329,10 +329,13 @@ def get_mach_header
 
     # Read just the file's magic number and check its validity.
     # @return [Fixnum] the magic
+    # @raise [MachO::NotAMachOError] if the file is too small to have valid magic
     # @raise [MachO::MagicError] if the magic is not valid Mach-O magic
     # @raise [MachO::FatBinaryError] if the magic is for a Fat file
     # @private
     def get_and_check_magic
+      raise NotAMachOError.new("file is too small to be a Mach-O") if @raw_data.size < 4
```

Comment:
This makes sense.

As a side note, do you know if `clang`/any part of the OS X toolchain actually ever produces just the Mach-O header (with `ncmds` and `sizeofcmds` = 0) and nothing else?